server.js

```diff
@@ -0,0 +1,64 @@
+const path = require('path');
+const express = require('express');
+const bearerToken = require('express-bearer-token');
+const proxy = require('http-proxy-middleware');
+const SDK = require('agile-sdk');
+const compression = require('compression');
+
+const app = express();
+const agile = SDK({
+  api: '/',
+  idm: 'http://agile-security:3000'
+})
+
+const authMiddleware = function (req, res, next) {
+  if (!req.token) {
+    return res.status(401).send('Authentication failed: Please provide a token');
+  }
+
+  agile.tokenSet(req.token);
+
+  agile.idm.user.getCurrentUserInfo()
+  .then(() => {
+    agile.tokenDelete();
```

Comment:
I am not sure, but this looks like there could be side effects when there is more than one request coming in? I know that the node.js process does not run in parallel, but still, if you get more than one request that is sleeping due to some async call like getCurrentUserInfo and get another request through the API this would mean that you are using another person's token right?

Reply:
Hmm, good find, I actually think it's okay because everything up until the token is used is synchronous, so we won't deal with another request. When the http request is made though other requests could be handled by and the token could change. So it will work in this case but it's not very robust. Let's instead instantiate a new `agile` object for each request that way there isn't any chance of that happening.